frontend/src/components/viewers/MetricsVisualizations.tsx

```diff
@@ -345,40 +345,54 @@ type ConfidenceMetric = {
 };
 
 interface ConfidenceMetricsSectionProps {
-  artifact: Artifact;
+  artifacts: Artifact[];
 }
 
-function ConfidenceMetricsSection({ artifact }: ConfidenceMetricsSectionProps) {
-  const customProperties = artifact.getCustomPropertiesMap();
-  const name = customProperties.get('display_name')?.getStringValue();
+export function ConfidenceMetricsSection({ artifacts }: ConfidenceMetricsSectionProps) {
+  const customPropertiesList = artifacts.map(artifact => artifact.getCustomPropertiesMap());
+  const names = customPropertiesList.map(customProperties =>
+    customProperties.get('display_name')?.getStringValue(),
```

Comment:
Should we filter out the artifact that doesn't have `display_name`?

Reply:
We didn't filter out the artifacts without `display_name` in the two-level dropdown - is there any reason for that behavior to change here? (I could filter out the artifacts whose corresponding run does not have a name in the next PR, as I did with Confusion Matrices, HTML, and Markdown.) This would also result in a change of behavior on the run detail page which uses this widget to display a single ROC curve, though that may be open to change.

Reply:
Got it, I remembered that we are using `artifact ID` in case of `artifact display_name` is empty for other metrics type. So I am curious about how we handle the `display_name` missing case here. Feel free to defer to future PRs as you mentioned that you are going to replace the current logic here with a table in the future.

Reply:
Yes, this is a good question. The long-term solution (which includes changes from subsequent PRs) will be to display the `display_name` of the run, execution, and artifact in the filter table referenced in my design doc (internal) and the PR Notes section; when this is not available, I will use the same replacement method as described here: https://github.com/kubeflow/pipelines/pull/7943#discussion_r911308942.

However, despite collecting all of the name information, this PR currently just uses the first name (when only one ROC Curve is present). This logic will be slightly modified in subsequent PRs as the table will already hold the relevant name information, but I want that the table display to be conditional on the use case (as the Run Details page will only have one ROC Curve on each artifact visualization, and thus should not have a table display).

Currently, the ROC Curve simply does not display any name if the name is not available, so I kept that same behavior; I can see how that would be confusing though, so I replaced it at least with a `-` at https://github.com/kubeflow/pipelines/pull/8046/commits/34cc7525f0991065f25bcb2dd1eb04bfc63fce81 (I am happy to use an ID instead, my current plan was just to do that for the filter table as in the case of a ROC curve the artifact being displayed is already clear), and added a `TODO` to clarify that this naming behavior would be further updated when the filter table is added.

Please let me know if I should make further changes on this logic (such as only fetching the first artifact name), and I'm happy to add `artifact ID` logic in this PR instead of the next.

Reply:
Sounds great! It is a good thing to point out that ROC Curve visualization doesn't display metric with empty name. Your changes make sense to me in terms of using artifact ID as replacement!